src/function/table/arrow.cpp

```diff
@@ -294,7 +294,9 @@ void ArrowTableFunction::ArrowScanFunctionParallel(ClientContext &context, const
 }
 
 idx_t ArrowTableFunction::ArrowScanMaxThreads(ClientContext &context, const FunctionData *bind_data_p) {
-	return context.db->NumberOfThreads();
+	auto &bind_data = (const ArrowScanFunctionData &)*bind_data_p;
+	idx_t rows_per_thread = 100000;
```

Comment:
Hardcoded constant?

Reply:
I can also pragma it up.
Or is there another preferred way?

Reply:
How about an optional parameter to the arrow scan function?

Reply:
sure, that sounds good, will do it later today :-)